tests/test_module.py

```diff
@@ -170,6 +170,22 @@ def test_entity_hex_euro_fail():
     assert "€" not in root[0].text
 
 
+# ############################# Text Content ############################# #
+
+
+def test_text_iterator():
+    html = "<html><body>sample text content</body></html>"
+    root = quick_parsehtml(html)
+    body = root.find(".//body")
+    assert "".join(body.itertext()) == "sample text content"
+
+def test_text_iterator_unclosed_tag():
+    html = "<html><body><span>hello <div>to <div>the <div>world!</span></body><footer>unrelated</footer></html>"
```

Comment:
Note that this test coverage is intentionally designed not to make any assumptions about the tree/node structure built in-memory by the parser.. because I'm not sure about the implications yet.

(for example: `div<div<div>>>` (nested nodes) is one possible in-memory representation, and `div<>div<>div<>` (sibling nodes) is another) -- either of those produce the same `itertext` results in _this_ test case, but it's possible to imagine representations where [descendent text content](https://dom.spec.whatwg.org/#concept-descendant-text-content) would be different depending on the parse tree)

Reply:
...and it turns out, the tree structure produced by `lxml` when 'implied end tags' are generated depends on the elements encountered! (by design)

A series of unclosed `div` elements will turn into nested nodes, whereas a series of unclosed `li` items will result in sibling nodes.

It's all spec'd in the whatwg HTML5 spec, but it's kinda complicated.  Both the current node and the previous node are relevant when deciding how to emit implied closing tags, and there are custom rules for various different elements.

Uncertain what to do here - I like that this library is simple and fast.  The reason I'm contributing is to see whether ScrapingHub's `extruct` library could use `htmlement` instead of `lxml`, allowing me to build pure-Python containers for `openculinary`.

Recent context here: https://github.com/scrapinghub/extruct/pull/163#issuecomment-1287166299 (and longer story earlier in that same issue)

In short: some of the test cases in `extruct` relate to tree-order traversal of parsed HTML nodes for text content retrieval, which is what led me to all this.

Reply:
Yeah that is a bit of an issue alright. While Htmlement's approach is fast, it's also very naive, deliberately so. 
I can add more logic to catch some of the more common closing tag issues, but it would never be fully compliant.